src/monster.h

```diff
@@ -288,13 +301,14 @@ class monster : public Creature, public JsonSerializer, public JsonDeserializer
         // Get torso - monsters don't have body parts (yet?)
         body_part get_random_body_part( bool main ) const override;
 
-        /** Resets a given special to its monster type cooldown value, an index of -1 does nothing. */
-        void reset_special(int index);
-        /** Resets a given special to a value between 0 and its monster type cooldown value.
-          * An index of -1 does nothing. */
-        void reset_special_rng(int index);
-        /** Sets a given special to the given value, an index of -1 does nothing. */
-        void set_special(int index, int time);
+        /** Resets a given special to its monster type cooldown value */
+        void reset_special(std::string special_name);
+        /** Resets a given special to a value between 0 and its monster type cooldown value. */
+        void reset_special_rng(std::string special_name);
+        /** Sets a given special to the given value */
+        void set_special(std::string special_name, int time);
+        /** Sets the enabled flag for the given special to false */
+        void disable_special(std::string special_name);
```

Comment:
Any reasons why those functions here require a copy (and not a const reference) of the special name?